modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Composed.kt

```diff
@@ -192,7 +200,11 @@ interface ComposedInvariantContravariant<F, G> : Invariant<Nested<F, G>> {
     fun G(): Contravariant<G>
 
     override fun <A, B> Kind<Nested<F, G>, A>.imap(f: (A) -> B, g: (B) -> A): Kind<Nested<F, G>, B> =
-        F().run { unnest().imap({ ga -> G().run { ga.contramap(g) } }, { gb -> G().run { gb.contramap(f) } }) }.nest()
+        F().run {
+            val fl: (Kind<G, A>) -> Kind<G, B> = { ga -> G().run { ga.contramap(g) } }
+            val fr: (Kind<G, B>) -> Kind<G, A> = { gb -> G().run { gb.contramap(f) } }
+            unnest().imap(fl, fr)
```

Comment:
Sorry, my bad. Take these out of the run block I mean.

Reply:
👌